packages/adapter/src/lib/_Types.ts

```diff
@@ -0,0 +1,402 @@
+    subscribable?: boolean;
+    /** compact group instance if running in compact mode */
+    compactInstance?: number;
+    /** if desired to have oStates. This is a list with all states values, and it will be updated automatically. */
+    states?: boolean;
+    /** if desired to have oObjects. This is a list with all states, channels and devices of this adapter, and it will be updated automatically.*/
+    objects?: boolean;
+    /** instance number of adapter */
+    instance?: number;
+    /** adapter directory name */
+    dirname?: string;
+    /** flag which defaults to true - if true, adapter warns if states are set without a corresponding existing object */
+    strictObjectChecks?: boolean;
+    /** If true runs in compact mode */
+    compact?: boolean;
+    /** configuration of the connection to controller */
+    config?: AdapterOptionsConfig;
+    /** name of the adapter. Must be exactly the same as directory name. */
+    name: string;
+    /** If true, the systemConfig (iobroker.json) will be available in this.systemConfig */
+    systemConfig?: boolean;
+    /** callback function (id, obj) that will be called if object changed */
+    objectChange?: ioBroker.ObjectChangeHandler;
+    /** callback function (id, obj) that will be called if state changed */
+    stateChange?: ioBroker.StateChangeHandler;
+    /** callback function (id, file) that will be called if file changed */
+    fileChange?: ChangeFileFunction;
+    /** callback to inform about new message the adapter */
+    message?: ioBroker.MessageHandler;
+    /** callback to stop the adapter */
+    unload?: ioBroker.UnloadHandler;
+    /** called when adapter is ready */
+    ready?: ioBroker.ReadyHandler;
+    /** called on reconnection to DB */
+    reconnect?: () => VoidLikeCallback;
+    /** Handler to handle uncaught exceptions, return true if no further handling required */
+    error?: ioBroker.ErrorHandler;
+}
+
+export type AdapterConfig = AdapterOptions | ioBroker.InstanceObject;
+
+export interface AdapterOptionsConfig {
+    log: {
+        level: ioBroker.LogLevel;
+    };
+}
+
+export interface InternalFormatDateOptions {
+    dateObj: string | Date | number;
+    isDuration: boolean;
+    _format?: string;
+}
+
+export interface AliasDetails {
+    source: AliasDetailsSource | null;
+    targets: AliasTargetEntry[];
+}
+
+export interface AliasDetailsSource {
+    min?: number;
+    max?: number;
+    type: string;
+    unit?: string;
+}
+
+export interface AliasTargetEntry {
+    alias: ioBroker.StateCommon['alias'];
+    id: string;
+    pattern: string;
+    type: string;
+    max?: number;
+    min?: number;
+    unit?: string;
+}
+
+export interface GetUserGroupsOptions {
+    user: `system.user.${string}`;
+    [other: string]: any;
+}
+
+export type CheckStateCommand = 'getState' | 'setState' | 'delState';
+
+export type VoidLikeCallback = Promise<void> | void;
```

Comment:
```suggestion
export type MaybePromise<T> = Promise<T> | T;
```
and replace `VoidLikeCallback` with `MaybePromise<void>`.

Or we can make it non-generic, but I don't like using "Callback" for a return type that is not a callback.